### PR TITLE
Post errors to datadog

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

This PR adds monitoring to [Datadog](https://datadoghq.eu/).

### Testing strategy

- [x] I ensured my code is tested: unit
- [x] I took care to mention breaking changes and extra libs if appropriate: [serialize-error](https://www.npmjs.com/package/serialize-error)

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "name": "Coorpacademy",
   "displayName": "Coorpacademy",
   "assistanceEmail": "assistance@coorpacademy.com",
-  "devToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJob3N0IjoiaHR0cHM6Ly9tb2JpbGUtc3RhZ2luZy5jb29ycGFjYWRlbXkuY29tIiwidXNhZ2UiOiJtb2JpbGUiLCJ1c2VyIjoiNWM3NTFhYTY4ZjI3MTMwMDFiMzBlMWIzIiwiZ3JhbnRzIjp7Im1vb2MiOnsiZ3JhbnRzIjp7Im1vYmlsZSI6eyJyb2xlcyI6WyJhZG1pbiIsImNvYWNoIiwidXNlciIsImdvZG1vZGUiXX19fX0sImlhdCI6MTU3MDAxOTU4MywiZXhwIjoxNjAwMDE5NTgzLCJpc3MiOiJjb29ycGFjYWRlbXktand0In0.mjp6E0WjE3IzWcJAGkP5PpWpML7Asyi_nw4Fk_uyzx0"
+  "devToken": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJob3N0IjoiaHR0cHM6Ly9tb2JpbGUtc3RhZ2luZy5jb29ycGFjYWRlbXkuY29tIiwidXNhZ2UiOiJtb2JpbGUiLCJ1c2VyIjoiNWM3NTFhYTY4ZjI3MTMwMDFiMzBlMWIzIiwiZ3JhbnRzIjp7Im1vb2MiOnsiZ3JhbnRzIjp7Im1vYmlsZSI6eyJyb2xlcyI6WyJhZG1pbiIsImNvYWNoIiwidXNlciIsImdvZG1vZGUiXX19fX0sImlhdCI6MTU3MDAxOTU4MywiZXhwIjoxNjAwMDE5NTgzLCJpc3MiOiJjb29ycGFjYWRlbXktand0In0.mjp6E0WjE3IzWcJAGkP5PpWpML7Asyi_nw4Fk_uyzx0",
+  "datadogToken": "ac5ed0a8f219e7e4113ed94c17cff885"
```
Comment (VincentCATILLON) [inline app.json]: should be discussed to see if it's relevant to have it in this file

File: src/components/authentication.js [hunk 2]
```diff
@@ -94,6 +96,9 @@ const Authentication = ({
         </View>
         <Space type="large" />
         <View style={styles.footer}>
+          <Text>Environment: {getBuildEnvironment()}</Text>
```
Comment (VincentCATILLON) [inline src/components/authentication.js]: temporary

File: src/components/authentication.js [hunk 3]
```diff
@@ -94,6 +96,9 @@ const Authentication = ({
...
+          <Text>Type: {getBuildType()}</Text>
+          <Text>Flavor: {getBuildFlavor()}</Text>
```
Comment (VincentCATILLON) [inline src/components/authentication.js]: temporary

File: src/components/authentication.js [hunk 4]
```diff
@@ -94,6 +96,9 @@ const Authentication = ({
         </View>
         <Space type="large" />
         <View style={styles.footer}>
+          <Text>Environment: {getBuildEnvironment()}</Text>
+          <Text>Type: {getBuildType()}</Text>
```
Comment (VincentCATILLON) [inline src/components/authentication.js]: temporary

File: app.json (hunk 1 above)
Comment (Fausto95) [reply app.json:6]: Well, this surely shouldn't be here IMO

Comment (AdrieanKhisbe) [reply app.json:6]: Pour rappel on veut pouvoir renouveller le token sans avoir à forcer le redeploiement du parc applicatif

File: scripts/update-version.js [hunk 5]
```diff
@@ -1,17 +1,35 @@
...
+  throw new Error(`Unsupported build flavor: ${buildFlavor}`);
+}
+
+const version: Version = {
+  branch: shell.exec('git rev-parse --abbrev-ref HEAD', {silent: true}).replace('\n', ''),
+  commit: shell.exec('git log -n 1 --pretty=format:"%h"', {silent: true}),
+  tag:
+    shell.exec('git describe --tags --abbrev=0', {silent: true}).replace('\n', '') ||
+    packageJson.version,
+  // $FlowFixMe this string is tested above
```
Comment (AdrieanKhisbe) [inline scripts/update-version.js:26]: je vois pas l'interer de mettre flow si c'est pour faire plus de flowFixMe que de type

File: src/modules/datadog.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    type?: BuildType,
+    flavor?: BuildFlavor
+  },
+  platform?: string,
+  brand?: string,
+  userId?: string
+|};
+
+export const getBuildEnvironment = (): BuildEnvironment => {
+  if (__TEST__) {
```
Comment (AdrieanKhisbe) [inline src/modules/datadog.js:43]: coule hold in 3 lines :grin:

**Pull request merged**